vignettes/how-to/use-shinymanager.Rmd

```diff
@@ -0,0 +1,52 @@
+---
+title: "How-to: Use shinymanager"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{How-to: Use shinymanager}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+To use [`shinymanager`](https://datastorm-open.github.io/shinymanager/) in Rhino
+you will need to setup a [legacy entrypoint](https://appsilon.github.io/rhino/reference/app.html#legacy-entrypoint)
+in your `rhino.yml`:
+
+```yml
+legacy_entrypoint: box_top_level
+```
+
+After adding `shinymanager` to your [dependencies](https://appsilon.github.io/rhino/articles/how-to/manage-r-dependencies.html)
+you can use it as follows:
+
+```r
+box::use(
+  shiny,
```

Comment:
Why do you think that this is wrong? Rhino-box tandem is against using `library` - which causes a global import of a package. Loading a package with `box` without specifying exact functions is still a valid way of utilizing `box` encapsulation:
1. The package is available only for this particular script.
2. Functions are available only with an explicit call `package_name$`
3. It is easier to read the code - you know exactly where the particular function comes from (e.g. `dplyr$filter` - you know it is from `dplyr`, you don't need to get to the top of the script to check if this is `filter` from `dplyr` or maybe from `MASS`.

Using this approach you can avoid the problem of dozens of exported functions listed in `box` import. In fact, we should encourage users to use this approach when more functions from a package are used in a script. Exactly the same as you can do this in Python.
Finally, we are already utilizing this type of import when importing scripts when importing a Rhino module.

Reply:
Okay, I may have be too adamant about this. Let me explain what I meant:

- I agree that loading a package without specifying exact functions with `box` is better than loading a package without specifying exact functions with `library`
- I agree this makes sense in some cases, for example for packages which import a lot of functions (for example `shiny` or `ggplot2`)
- I also don't have a strong opinion on `dplyr$filter` improving readability, because this is almost the same as `dplyr::filter` and readability may depend on the person using the code: it may be OK for some developers and not OK for others.

Still, in my opinion we should still consider using construction like `shinymanager[fun1, fun2, fun2]` for educational purposes in vignettes because `box` is not an industry standard and it's quite probable that our vignettes will be encountered by developers who never used it.
Because of that, it would be good to familiarize readers of our documentation with an idea of loading just a subset of package namespace so they are more likely to use it when the need arises.

This is in line with my comment about repeating the warnings about the passwords. I know we already inform users about those good practices in different places, but we frankly can't be sure if they read it since we don't know what will be the order of encountering our documentation and if a person reads it all. So I believe reiterating in multiple places make our docs more 'bulltet-proof' and useful which can improve the adoption in the long term.

Reply:
I agree that pointing to the article about `box` or in general, cross-referencing documentation topics is a good practice. But here, regarding the ways of using the import feature of `box` - __both ways (explicit list of functions and importing a package and referring to it via $) are good practices__. One is not worse than the other, but should be used in slightly different scenarios - if there are just a few functions, list them, if there are more - import just a package. From what I have seen, people tend to just list functions and this quickly becomes a nightmare. I see showing that there is another way extremely beneficial.

Reply:
I had a short discussion with @Leszek-Sieminski and we agreed to keep the whole-package imports here.